ext/ffi_c/extconf.rb

```diff
@@ -8,7 +8,11 @@ def system_libffi_usable?
     # We need pkg_config or ffi.h
     libffi_ok = pkg_config("libffi") ||
         have_header("ffi.h") ||
-        find_header("ffi.h", "/usr/local/include", "/usr/include/ffi")
+        find_header("ffi.h",
+                    "/usr/local/include",
+                    "/usr/include/ffi",
+                    "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/ffi",
+                    "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/ffi")
```

Comment:
Wouldn't using `xcrun --sdk macosx --show-sdk-path` to get the SDK path and then append `/usr/include/ffi` be more future-proof?

Reply:
Shelling out to `xcrun` can cause some unexpected behaviour unless you catch those cases and I'm not sure I'd recommend it.

Namely that if you haven't properly configured your development setup on macOS you'll get a graphical pop-up smack back in the middle of your screen asking if you'd like to install the CLT, and if you haven't accepted the Xcode/CLT license `xcrun` just errors out telling you to do that first. They are edge cases but, people love to stumble upon edge cases.

Reply:
Compiling won't work until the CLT are installed anyway, so maybe it's fine to get the popup then if the user still did not install command-line tools?

Re the license, we could validate the output and only use it if it exists as a directory.

Reply:
I opened #765 for a version using `xcrun`. I don't have a strong opinion to any of both. And I don't have macOS, so maybe this is something for someone who can actually try the patch out :smile: